### DISCUSSION TITLE
Force Window to Open on Active Monitor(s)

### DISCUSSION THREAD

**Discussion by rthndr** [sumatrapdfreader/sumatrapdf#3776] [Q&A]
I switch between single and multi monitor setups. When a file was opened on a secondary display during a multi-monitor session and then reopened on a single monitor session, the only way to view the file is to right click the taskbar and cascade/stack windows. How can I force the files to open on the the active display in a single monitor session?

For context, the file will show in the taskbar but clicking the button doesn't bring the window into view. It's like the window is still "off screen" except there isn't another screen for it to be on.

**Comment by GitHubRulesOK** [comment]
That is fairly common for many apps that keep track of monitors, then the monitor is disconnected. So in the past I have had to ensure when unplugging laptop from docking port extender, that all windows are on main screen before disconnect. It should generally not be an issue when using a tabbed session except when there was a second window opened.

If you suspect 2 windows are open then right click taskbar icon and select close all windows, then on restart SumatraPDF may in that event only reopen the main one ! but if that was off screen its trickier to return without you edit the settings file. For example if `WindowPos = 1825 396 1034 349` in `SessionData [
	[`  then set that one AFTER closing SumatraPDF to  `WindowPos = 0 396 1034 349` 
This is fiddly  as you may need to reduce multiple `WindowsPos` entries not just one, and thus agreed a workaround to the fact that the remembered state is out of single screen boundary, but the alternative would be to not remember opened files and their positions .

**Comment by xChavetaz** [comment]
Any solution for this problem? currently many applications are on more than one monitor and when you disconnect one, they all go to active. Except SumatraPDF, it is stuck to the old location of the disconnected monitor...
Is it possible to deactivate the last position of the application? I understand that this way it will always be displayed on the active monitor.

**Reply by GitHubRulesOK** [reply]
it should be "easy" to script a reset function based on where your settings are as all it needs is set a 0
In my settings the current value is WindowPos = -1909 -... as it is a monitor on the left and 1920 wide
the problem is as highly variable you will need a personal value for example 
WindowPos= 0 600 400 900

AI suggests use reset.cmd in same folder as settings
```
 @echo off
cd /d "%~dp0"
REM ================================
REM User‑editable values
REM ================================
set S1=0
set S2=0
set S3=900
set S4=600

REM ================================
REM Process file using PowerShell
REM ================================
powershell -NoLogo -NoProfile -Command ^
  "(Get-Content 'SumatraPDF-settings.txt' -Encoding UTF8) -replace 'WindowPos\s*=\s*-?\d+\s+-?\d+\s+-?\d+\s+-?\d+', 'WindowPos = %S1% %S2% %S3% %S4%' | Set-Content 'SumatraPDF-settings.txt' -Encoding UTF8"

pause
```

**Comment by kjk** [comment]
@rthndr @xChavetaz Please try 17831 pre-release and let me know if it fixes the issue. I've made a fix but have no way to test it.

You can either upgrade from withing (`Ctrl + k`, `Check for updates`) or download from:
https://www.sumatrapdfreader.org/dl/prerel/17831/SumatraPDF-prerel-64.exe
https://www.sumatrapdfreader.org/dl/prerel/17831/SumatraPDF-prerel-64-install.exe

**Comment by GitHubRulesOK** [comment]
@kjk when 17831 is installed  and second monitor switched off the apps all naturally fall back to primary'
That now includes sumatraPDF although I dont know if it failed under different conditions in the past.
If I switch secondary monitor on again NO app will be seen active on that as they were automatically repositioned on primary  Thus I feel that behaviour is "as expected"

When I shutdown with 2 monitors active SumatraPDF closes correctly on second monitor

Then on start with only one monitor it should be seen  in correct screen SADLY it is not although I can see the edge of a new window at edge of screen the startup settings are in the OLD position off primary.

**Comment by GitHubRulesOK** [comment]
@kjk because ctrl K+  cannot be seen
There to my mind 2 options one is the taskbar icon include a reposition on active monitor
 OR my suggested reset the win positions to a default say 600 x 600 for my small factor screen (works for me)

**Comment by xChavetaz** [comment]
I perform the following test: with 2 monitors connected (extended mode), I open a pdf on monitor 1, move it to monitor 2 and close the application and disconnect monitor 2.

With the official version Sumatrapdf opens offscreen. 
With this new version it opens on monitor 1.

maybe there are other ways to do it, but in my opinion, it is much more usable than the previous version (you had to stand on the application in the taskbar, right click on the window, select Move and then with the keys direction move the window to monitor 1).

Many thanks!

**Reply by GitHubRulesOK** [reply]
@xChavetaz 
OK there are 3 ways :-) (I had not thought of the obvious)
<img width="235" height="250" alt="image" src="https://github.com/user-attachments/assets/67930860-f4b9-43e9-a810-9ae1922ca2c8" />
